bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -61,68 +74,121 @@ public void initialize() {
 
         EnergyMeterConfig config = getConfigAs(EnergyMeterConfig.class);
 
-        int port = (config.getPort() == null) ? EnergyMeter.DEFAULT_MCAST_PORT : config.getPort();
-        energyMeter = new EnergyMeter(config.getMcastGroup(), port);
+        int port = (config.getPort() == null) ? PacketListener.DEFAULT_MCAST_PORT : config.getPort();
+
         try {
-            energyMeter.update();
+            PacketListener thingListener = listenerRegistry.getListener(config.getMcastGroup(), port);
+            serialNumber = config.getSerialNumber();
+            if (serialNumber == null) {
+                if (!thing.getProperties().containsKey(Thing.PROPERTY_SERIAL_NUMBER)
+                        || (serialNumber = retrieveSerialNumber(thing.getProperties())) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Meter serial number missing");
+                    return;
+                }
+
+                // copy serial number from thing properties into config
+                Map<String, Object> newConfig = editConfiguration().getProperties();
+                newConfig.put("serialNumber", serialNumber);
+                Thing thing = editThing().withConfiguration(new Configuration(newConfig)).build();
+                updateThing(thing);
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            } else {
+                if (!thing.getProperties().containsKey(Thing.PROPERTY_SERIAL_NUMBER)) {
+                    Map<String, String> props = editProperties();
+                    props.put(Thing.PROPERTY_VENDOR, "SMA");
+                    props.put(Thing.PROPERTY_SERIAL_NUMBER, serialNumber);
+                    updateProperties(props);
+                }
+            }
 
-            updateProperty(Thing.PROPERTY_VENDOR, "SMA");
-            updateProperty(Thing.PROPERTY_SERIAL_NUMBER, energyMeter.getSerialNumber());
-            logger.debug("Found a SMA Energy Meter with S/N '{}'", energyMeter.getSerialNumber());
+            logger.debug("Activated handler for SMA Energy Meter with S/N '{}'", serialNumber);
+
+            this.updateInterval = TimeUnit.SECONDS
+                    .toMillis(config.getPollingPeriod() == null ? 30 : config.getPollingPeriod());
+            timeoutFuture = scheduler.scheduleWithFixedDelay(this, updateInterval * 2, updateInterval,
+                    TimeUnit.MILLISECONDS);
+            this.listener = thingListener;
+            thingListener.addPayloadHandler(this);
+            thingListener.open();
+            logger.debug("Listening to meter updates and publishing its measurements every {}ms for '{}'",
+                    updateInterval, getThing().getUID());
+            // we do not set online status here, it will be set only when data is received
```

Comment:
initialize should always set a thing status before returning.
Are you certain that you will receive a data and immediately after ? If not, set the thing status to UNKNOWN and then to ONLINE when a data will be received.

Reply:
Few lines above there is a scheduled task which will mark thing offline if packet does not arrive within two poll cycles.